sbc_compassion/models/import_letters_history.py

```diff
@@ -274,10 +306,9 @@ def _analyze_pdf(self, pdf_data, file_name):
             # pylint: disable=invalid-commit
             self._cr.commit()
         except Exception:
-            message = f"Couldn't import file {file_name}"
-            self.env.user.notify_danger(message)
-            logger.error(
-                message,
-                exc_info=True,
-            )
+            self.import_completed = False
+            self.failed_file_name = file_name
+            self.state = "failed"
+            self.env.user.notify_danger(f"Couldn't import file {file_name}")
+            self._compute_state = False
```

Comment:
A call to the write method is preferable when changing many values at the same time. Also assigning the compute method seems wrong.
```suggestion
            self.write({
                "import_completed": False,
                "failed_file_name": file_name,
                "state": "failed"
            })
            self.env.user.notify_danger(f"Couldn't import file {file_name}")
```

Reply:
when I remove the line "self._compute_state = False" the import is recalculated and the import state is set to ready instead of failed

Reply:
Can you then please put this as a comment above the line in order to explain the reason of this line? Otherwise won't understand and may remove it in the future.